### PR TITLE
OpenGL visualization improved

### PR DESCRIPTION
**Pull request by KonradBreitsprecher**
- Improved mouse control: Left button: Rotate system, Right Button: Look around, Wheel: Zoom
- Improved lightning: A spotlight on the camera illuminates the point of view.
- Decoupled controls from visualizer.update(), making it possible to inspect the system if espresso integration is blocking.
- Fixed visualizer sample scripts
- Extended doc

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: src/python/espressomd/visualizationOpenGL.pyx [hunk 1]
```diff
@@ -68,19 +74,14 @@ class openGLLive(object):
...
                 raise ValueError(
-                    key + 'is not a valid visualization property')
+                    key + ' is not a valid visualization property')
             else:
                 self.specs[key] = kwargs[key]
 
         #DEPENDENCIES
-        #if not 'EXTERNAL_FORCES' in espressomd.features():
-        IF not EXTERNAL_FORCES:
+        if not 'EXTERNAL_FORCES' in espressomd.features():
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualizationOpenGL.pyx]: that is a pyx file, so you can use the preprocessing

File: src/python/espressomd/visualizationOpenGL.pyx [hunk 2]
```diff
@@ -737,27 +762,57 @@ class openGLLive(object):
                      'background_color'][1], self.specs['background_color'][2], 1.)
 
         glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA)
-        glEnable(GL_BLEND)
+        #glBlendFunc(GL_SRC_ALPHA, GL_ONE)
+        #glDepthMask(GL_TRUE)
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualizationOpenGL.pyx]: can you remove commented code

File: src/python/espressomd/visualizationOpenGL.pyx [hunk 3]
```diff
@@ -1178,9 +1217,19 @@ class Camera(object):
...
-        self.camRot += (mousePos - mousePosOld) * self.lookSpeed
-        self.calcCameraDirections()
+    def rotateCamera(self, mousePos, mousePosOld, mouseButtonState):
+        dm = mousePos - mousePosOld
+
+        if mouseButtonState[GLUT_LEFT_BUTTON] == GLUT_DOWN:
+            self.camRotGlobal[1] += dm[0] * 0.1 * self.globalRotSpeed
+            self.camRotGlobal[0] += dm[1] * 0.1 * self.globalRotSpeed
+        elif mouseButtonState[GLUT_RIGHT_BUTTON] == GLUT_DOWN:
+        #    self.camRotGlobal[2] -= dm[0] * 0.2 * self.globalRotSpeed
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualizationOpenGL.pyx]: please remove unused code

**Comment by KonradBreitsprecher** [comment]
Seems like I've overwritten parts of io.rst while editing, I'm going to close this for now and fix it.
Thanks @KaiSzuttor for the review.

**Review by KaiSzuttor — APPROVED** [review]

**Comment by KonradBreitsprecher** [comment]
Fixes #1129

**Review by mkuron — APPROVED** [review]

**Pull request merged**